yt_dlp/extractor/zattoo.py

```diff
@@ -243,7 +294,67 @@ def _make_valid_url(tmpl, host):
 
 
 class ZattooIE(ZattooBaseIE):
-    _VALID_URL_TEMPLATE = r'https?://(?:www\.)?%s/watch/(?P<channel>[^/]+?)/(?P<id>[0-9]+)[^/]+(?:/(?P<recid>[0-9]+))?'
+    _VALID_URL_TEMPLATE = r'''(?x)
+                            https?://(?:www\.)?%s/
+                            (?:
+                                (?:
+                                    vod/movies/([A-Za-z0-9]+)|
+                                    (?:program|watch)/([^/]+)/(\d+)|
+                                    live/([^/]+)
+                                )|
+                                (?:
+                                    .+\?recording=([0-9]+)|
+                                    .+\?movie_id=([A-Za-z0-9]+)|
+                                    .+\?channel=([^&]+)(?:
+                                        &program=(\d+)
+                                    )?
+                                )
+                            )'''
+
+    _VALID_URL = _make_valid_url(_VALID_URL_TEMPLATE, ZattooBaseIE._HOST)
+    _TESTS = [{
+        'url': 'https://zattoo.com/channels/german?channel=srf_zwei',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/guide/german?channel=srf1&program=169860555',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/recordings?recording=193615508',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/tc/ptc_recordings_all_recordings?recording=193615420',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/vod/movies/7521',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/program/daserste/210177916',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/ondemand?movie_id=7521&term_token=9f00f43183269484edde',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/live/srf1',
+        'only_matching': True,
+    }]
```

Comment:
Add atleast one proper test

Reply:
It is not possible to add a proper test, like goggle already pointed out. But I could provide you with a (German) Zattoo account. Is any of the maintainers willing to test manually? (pukkandan told me, he does not have time to do so)

Reply:
> It is not possible to add a proper test,

There are other uses for `_TESTS` than just the automated tests. You can add a test that currently works and then add `'skip': True` to skip it in future automated tests

Reply:
@waldorfpatriot I'll have a look.
@pukkandan When I add a test, how do I run this test locally on my computer? I cannot use the `test_download` script because I don't know how to provide login credentials to that script...

Reply:
@waldorfpatriot Downloading from the "Guide" still works for me with this PR. Do you have a German account? I have only checked it with a Swiss account... If downloading from the 7-day-replay section doesn't work for you, would you be able to give your account credentials (you find my personal email on my profile)?

Reply:
I have added a test, but it still fails with this error message:
```
======================================================================
FAIL: test_Zattoo_all (__main__.TestDownload):
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/alex/Projects/repos/yt-dlp/test/test_download.py", line 281, in test_template
    getattr(self, f'test_{name}_{i}' if i else f'test_{name}')()
  File "/home/alex/Projects/repos/yt-dlp/test/test_download.py", line 234, in test_template
    assertGreaterEqual(
  File "/home/alex/Projects/repos/yt-dlp/test/helper.py", line 303, in assertGreaterEqual
    self.assertTrue(got >= expected, msg)
AssertionError: False is not true : Expected test_Zattoo_250380873.mp4 to be at least 9.77KiB, but it's only 685.00B 

----------------------------------------------------------------------
Ran 1 test in 3.052s
```
Downloading the whole video works fine...
Any idea why this happens?

I have also noticed that by default, the video will be taken from the HLS stream while the audio is from the DASH stream. Is there an easy fix for that?

Reply:
fragmented downloads cannot be md5 tested. Add a `'params': {'skip_download': 'm3u8'}` to the test to skip it

Reply:
One more thing I've noticed:
In the test
```
'url': 'https://zattoo.com/program/zdf/250380873',
        'info_dict': {
            'id': '250380873',
            'ext': 'mp4',
            'title': 'heute-show',
            'description': 'md5:413cf29b7a2f157455a4581aced7a3e9',
            'thumbnail': 'md5:62e70eb5cc3a2e203773b43be051196d',
            'creator': 'ZDF HD',
            'release_year': 2022,
            'episode': 'Folge 407',
            'categories': ['Séries'],
            'tags': ['Comédie']
        },
        'params': {'skip_download': 'm3u8'}
```
the `categories` and `tags` lists seem to depend on the chosen language of the Zattoo account. So if there is a way to exclude these from the test and still make the test pass, that would be great.

Reply:
```py
            'categories': 'count:1',
            'tags': 'count:1',
```